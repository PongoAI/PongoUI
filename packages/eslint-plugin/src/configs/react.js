/** @type {import("eslint").Linter.Config} */
const config = {
  extends: ['airbnb-typescript', 'prettier'],
  parser: '@typescript-eslint/parser',
  parserOptions: {
    project: 'tsconfig.json',
  },
  plugins: ['deprecation', 'import', '@typescript-eslint', 'jest', 'jsx-a11y', 'react', 'react-hooks'],
  settings: {
    'import/parsers': {
      '@typescript-eslint/parser': ['.ts', '.tsx'],
    },
    'import/resolver': {
      typescript: {
        alwaysTryTypes: true,
      },
    },
    'no-unused-expressions': 'off',
  },
  env: {
    browser: true,
    es6: true,
    node: true,
  },

  ignorePatterns: ['coverage', 'dist', 'dist-storybook', 'lib', 'lib-commonjs', '**/__snapshots__', 'node_modules'],
  rules: {
    'no-unused-vars': 'error',
    'no-use-before-define': 'off',
    '@typescript-eslint/no-use-before-define': ['error'],
    'import/no-extraneous-dependencies': ['error', { devDependencies: true }],
    'import/extensions': [
      'error',
      'ignorePackages',
      {
        js: 'never',
        jsx: 'never',
        ts: 'never',
        tsx: 'never',
      },
    ],
  },
};

module.exports = config;
