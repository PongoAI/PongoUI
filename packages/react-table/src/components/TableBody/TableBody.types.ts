import type { ComponentProps, ComponentState, Slot } from '@fluentui/react-utilities';

export type TableBodySlots = {
  /**
   * The root of the TableBody.
   */
  root: Slot<'tbody'>;
};

type TableBodyCommons = {};

export type TableBodyProps = ComponentProps<TableBodySlots> & TableBodyCommons;

export type TableBodyState = ComponentState<TableBodySlots> & TableBodyCommons;
