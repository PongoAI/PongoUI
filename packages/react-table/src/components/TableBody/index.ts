export * from './TableBody';
export * from './useTableBodyStyles';
export * from './TableBody.types';
export * from './useTableBody';
