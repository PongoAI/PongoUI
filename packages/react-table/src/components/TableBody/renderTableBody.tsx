import * as React from 'react';
import { getSlots } from '@fluentui/react-utilities';
import type { TableBodyState, TableBodySlots } from './TableBody.types';

export const renderTableBody = (state: TableBodyState) => {
  const { slots, slotProps } = getSlots<TableBodySlots>(state);

  return <slots.root {...slotProps.root} />;
};
