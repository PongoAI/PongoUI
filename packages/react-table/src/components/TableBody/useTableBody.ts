import * as React from 'react';
import { getNativeElementProps } from '@fluentui/react-utilities';
import type { TableBodyProps, TableBodyState } from './TableBody.types';

export const useTableBody = (props: TableBodyProps, ref: React.Ref<HTMLElement>): TableBodyState => {
  const state: TableBodyState = {
    components: {
      root: 'tbody',
    },
    root: getNativeElementProps('tbody', {
      ref,
      ...props,
    }),
  };

  return state;
};
