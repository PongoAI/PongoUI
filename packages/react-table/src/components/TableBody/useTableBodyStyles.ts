import { makeStyles, mergeClasses, shorthands } from '@griffel/react';
import type { TableBodyState } from './TableBody.types';
import { tokens } from '@pongo-ui/react-theme';

const useRootStyles = makeStyles({
  root: {
    display: 'table-row-group',
    verticalAlign: 'middle',
    ...shorthands.borderColor('inherit'),
    fontWeight: tokens.regular,
  },
});

export const useTableBodyStyles = (state: TableBodyState) => {
  const rootStyles = useRootStyles();

  state.root.className = mergeClasses(rootStyles.root, state.root.className);

  return state;
};
