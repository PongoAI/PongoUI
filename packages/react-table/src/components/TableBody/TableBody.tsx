import * as React from 'react';
import { useTableBody } from './useTableBody';
import { useTableBodyStyles } from './useTableBodyStyles';
import { renderTableBody } from './renderTableBody';
import type { TableBodyProps } from './TableBody.types';
import type { ForwardRefComponent } from '@fluentui/react-utilities';

/**
 * The TableBody component is used to display the main collection of data.
 */
export const TableBody: ForwardRefComponent<TableBodyProps> = React.forwardRef((props, ref) => {
  const state = useTableBody(props, ref);

  useTableBodyStyles(state);

  return renderTableBody(state);
});
TableBody.displayName = 'TableBody';
