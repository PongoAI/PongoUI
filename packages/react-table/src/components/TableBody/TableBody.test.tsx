import * as React from 'react';
import { TableBody } from './TableBody';
import { render } from '@testing-library/react';
import { isConformant } from '../../common/isConformant';

describe('TableBody', () => {
  isConformant({
    Component: TableBody,
    displayName: 'TableBody',
  });

  it('renders a basic Table', () => {
    const renderedComponent = render(<TableBody />);

    expect(renderedComponent.baseElement).toMatchSnapshot();
  });
});
