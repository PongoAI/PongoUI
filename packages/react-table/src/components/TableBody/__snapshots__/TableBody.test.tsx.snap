// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TableBody renders a basic Table 1`] = `
<body>
  <div>
    <tbody
      class="___1clrp98 f1tp1avn fmrv4ls fetgrfm fwsgm2r f1kc6foo f13jenf0 f1rd5vql"
    />
  </div>
</body>
`;
